src/UnstableApis.ts

```diff
@@ -83,6 +93,139 @@ export class UnstableApis {
+     * Gets the users in a group.
+     * @param {string} groupId The group ID of which to get the users.
+     * @return {Promise<any[]>} Resolves to an array of all the users in the group.
+     */
+    public async getGroupUsers(groupId: string): Promise<any[]> {
+        const response = await this.client.doRequest("GET", `/_matrix/client/r0/groups/${encodeURIComponent(groupId)}/users`);
+        return response["chunk"];
+    }
+
+    /**
+     * Gets the invited users of a group.
+     * @param {string} groupId The group ID of which to get the invited users.
+     * @return {Promise<any[]>} Resolves to an array of all the users invited to the group.
+     */
+    public async getGroupInvitedUsers(groupId: string): Promise<any[]> {
+        const response = await this.client.doRequest("GET", `/_matrix/client/r0/groups/${encodeURIComponent(groupId)}/invited_users`);
+        return response["chunk"];
+    }
+
+    /**
+     * Gets the rooms of a group.
+     * @param {string} groupId The group ID of which to get all the rooms.
+     * @return {Promise<any[]>} Resolves to an array of all the rooms of the group.
+     */
+    public async getGroupRooms(groupId: string): Promise<any[]> {
+        const response = await this.client.doRequest("GET", `/_matrix/client/r0/groups/${encodeURIComponent(groupId)}/rooms`);
+        return response["chunk"];
+    }
+
+    /**
+     * Accepts an invite to a group.
+     * @param {string} groupId The group ID of which to accept the invite of.
+     * @return {Promise<any>} Resolves when completed.
+     */
+    public async acceptGroupInvite(groupId: string): Promise<any> {
+        return this.client.doRequest("PUT", `/_matrix/client/r0/groups/${encodeURIComponent(groupId)}/self/accept_invite`, null, {});
+    }
+
+    /**
+     * Joins a group.
+     * @param {string} groupId The group ID to join.
+     * @return {Promise<any>} Resolves when completed.
+     */
+    public async joinGroup(groupId: string): Promise<any> {
+        return this.client.doRequest("PUT", `/_matrix/client/r0/groups/${encodeURIComponent(groupId)}/self/join`, null, {});
+    }
+
+    /**
+     * Leaves a group.
+     * @param {string} groupId The group ID of the group to leave.
+     * @return {Promise<any>} Resolves when completed.
+     */
+    public async leaveGroup(groupId: string): Promise<any> {
+        return this.client.doRequest("PUT", `/_matrix/client/r0/groups/${encodeURIComponent(groupId)}/self/leave`, null, {});
+    }
+
+    /**
+     * Sets the publicity of a group.
+     * @param {string} groupId The group ID to set the publicity of.
+     * @param {boolean} publicise If the group should be publicised.
+     * @return {Promise<any>} Resolves when completed.
+     */
+    public async setGroupPublicity(groupId: string, publicise: boolean): Promise<any> {
+        return this.client.doRequest("PUT", `/_matrix/client/r0/groups/${encodeURIComponent(groupId)}/self/update_publicity`, null, {
+            publicise,
```

Comment:
it kinda bothers me that the spec effectively forces UK English onto the world. This project is supposed to be Canadian English (despite not being mentioned), which is basically American English with more `u`s, but because the spec has decided to be UK English it forces the hand of this project a bit :|

`</rant about the spec>`

Reply:
So...should soru change anything?

Reply:
no, I'm just complaining about the spec.